### ISSUE TITLE
Unable to use SSL on macOS Catalina

### ISSUE THREAD

**Issue by fredol** [drud/ddev#2468]
After several reinstall, remove and clean install, I'm still unable to have SSL enabled dev host.

Each time I start a ddev project, I always receive this warning:

mkcert may not be properly installed, we suggest installing it for trusted https support, `brew install mkcert nss`, `choco install -y mkcert`, etc. and then `mkcert -install` 

and trying to access to dev instance via https gives me an exception.

- Host computer: macOS Catalina 10.15.6 fully up to date
- Docker Desktop version: 2.3.0.4
- ddev version information: 1.15.3

Thank in advance for your help

**Comment by rfay** [comment]
Have you done the required `mkcert -install` ? You don't mention that. 

If you *have*, please 
1. Provide the contents of the top part of your ~/.ddev/global-config.yaml (the projects list isn't needed, nor the comments)
2. `mkcert -uninstall && rm -r "$(mkcert -CAROOT)" && mkcert -install`
3. Try again.

**Comment by fredol** [comment]
Yes, done this several times unsuccessfully 

1. 
frederic:TYPO3-Composer frederic$ cat ~/.ddev/global_config.yaml
omit_containers: []
nfs_mount_enabled: false
instrumentation_opt_in: true
router_bind_all_interfaces: false
internet_detection_timeout_ms: 750
last_started_version: v1.15.3
mkcert_caroot: ""

2.
 frederic:Documents frederic$ ddev poweroff
 frederic:Documents frederic$ sudo mkcert -uninstall
Using the local CA at "/Users/frederic/Library/Application Support/mkcert" ✨
The local CA is now uninstalled from the system trust store(s)! 👋

frederic:Documents frederic$ sudo rm -r "$(mkcert -CAROOT)"
frederic:Documents frederic$ sudo mkcert -install
Created a new local CA at "/Users/frederic/Library/Application Support/mkcert" 💥
The local CA is now installed in the system trust store! ⚡️
The local CA is now installed in the Firefox trust store (requires browser restart)! 🦊

3.
frederic:Documents frederic$ cd TYPO3-Composer/
frederic:TYPO3-Composer frederic$ ddev start
Starting TYPO3-Composer... 
mkcert may not be properly installed, we suggest installing it for trusted https support, `brew install mkcert nss`, `choco install -y mkcert`, etc. and then `mkcert -install` 
Building db 
Building web 
Recreating ddev-TYPO3-Composer-db ... done
Recreating ddev-TYPO3-Composer-web ... done
Recreating ddev-TYPO3-Composer-dba ... done
 
ddev-router is up-to-date 
Generating AdditionalConfiguration.php file for database connection. 
Successfully started TYPO3-Composer 
Project can be reached at http://TYPO3-Composer.ddev.site http://127.0.0.1:32785

**Comment by rfay** [comment]
What does `mkcert -CAROOT` say?

It looks like it's not working at all. (your ~/.ddev/global_config.yaml says `mkcert_caroot: ""`.  Please show the results of 
```
mkcert -CAROOT
mkcert -install
```

Is mkcert in your $PATH?

**Comment by rfay** [comment]
You didn't mention how you installed ddev. With `brew tap drud/ddev && brew install ddev` ?

**Comment by fredol** [comment]
DDEV is installed with `brew tap drud/ddev && brew install ddev`

frederic:TYPO3-Composer frederic$ mkcert -CAROOT
/Users/frederic/Library/Application Support/mkcert

frederic:TYPO3-Composer frederic$ mkcert -install
Using the local CA at "/Users/frederic/Library/Application Support/mkcert" ✨
ERROR: failed to read the CA key: open /Users/frederic/Library/Application Support/mkcert/rootCA-key.pem: permission denied

mkcert location: /usr/local/bin/mkcert 

my PATH is 
PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Applications/Wireshark.app/Contents/MacOS

so this is fine for mkcert

**Comment by rfay** [comment]
Ah, you seem to have used sudo to run mkcert at some point. Don't do that.  Now your user (and ddev) can't run mkcert successfully.

Here's how to fix:
```
sudo mkcert -uninstall
sudo rm -rf "$(mkcert -CAROOT)"
mkcert -install
```

**Comment by fredol** [comment]
### **Yeah! It works.**

**Thank you very much for your support!**

I close the issue

**Comment by rfay** [comment]
If this ever happens to anybody, especially on macOS, WSL2, or Linux, `ls -l "$(mkcert -CAROOT)"` should show only files owned by your user, not root. If you see ownership by root, you can use the technique in https://github.com/drud/ddev/issues/2468#issuecomment-677705956 to fix it.

**Comment by gambry** [comment]
My user on MacOS is not an administrator and running `mkcert -install` was asking me for a sudo password. Trying to install as administrator failed too.

I noticed the root CA for the two users was different so I tried the [Installing the CA on other systems](https://github.com/FiloSottile/mkcert#installing-the-ca-on-other-systems) described in mkcert documentation and copied over rootCA*.pem from my user CAROOT to admin's one and then I ran `mkcert -install`. This fixed the issue for me.
